src/main/java/com/gotocompany/depot/maxcompute/client/insert/session/StreamingSessionManager.java

```diff
@@ -0,0 +1,58 @@
+package com.gotocompany.depot.maxcompute.client.insert.session;
+
+import com.aliyun.odps.tunnel.TableTunnel;
+
+import com.aliyun.odps.tunnel.TunnelException;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+
+public final class StreamingSessionManager {
+
+    private final LoadingCache<String, TableTunnel.StreamUploadSession> sessionCache;
+
+    private StreamingSessionManager(CacheLoader<String, TableTunnel.StreamUploadSession> cacheLoader,
+                                   MaxComputeSinkConfig maxComputeSinkConfig) {
+        sessionCache = CacheBuilder.newBuilder()
+                .maximumSize(maxComputeSinkConfig.getStreamingInsertMaximumSessionCount())
+                .build(cacheLoader);
+    }
+
+    public static StreamingSessionManager nonParititonedStreamingSessionManager(TableTunnel tableTunnel, MaxComputeSinkConfig maxComputeSinkConfig) {
```

Comment:
Usage will look like StreamingSessionManager.nonParititonedStreamingSessionManager repeatative

I suggest to use `createNonPartioned` or `newNonPartioned` or `newPartionedInstance`